conjur/init_logic.py

```diff
@@ -0,0 +1,107 @@
+  DEFAULT_PROMPT_RESPONSE=True
+
+  def __init__(self):
+    pass
+
+  @staticmethod
+  def connect_to_socket(hostname, port):
+    """
+    Method for opening a socket with the Conjur server
+    """
+    try:
+      if port == None:
+        port = 443
+
+      ctx = SSL.Context(method=SSL.TLSv1_2_METHOD)
+      # We set the following to false because we want to fetch the certificates and not use them to validate yet
+      ctx.check_hostname = False
+      ctx.verify_mode = False
+
+      sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+      sock = SSL.Connection(context=ctx, socket=sock)
+      sock.connect((hostname, 443))
+      sock.do_handshake()
+    except Exception as e:
+      raise Exception(f"Unable to retrieve certificate from {hostname}:{port}")
+
+    logging.info("TLS Connection established. " \
+             "Fetching certificate from server...")
+    return sock
+
+  def get_certificate(self, hostname, port, sock):
+    """
+    Method for connecting to Conjur to fetch the certificate chain
+    """
+    from OpenSSL.crypto import dump_certificate, FILETYPE_PEM
+    try:
+      chain = sock.get_peer_cert_chain()
+      cert = chain[0]
+      fp = cert.digest("sha1").decode("utf-8")
+
+      # Get the peer chair and format to make it easier to later write to a file
+      readable_certificate=''
+      for (idx, cert) in enumerate(chain):
+          readable_certificate+=str(dump_certificate(FILETYPE_PEM, cert),  "utf-8")
+
+      logging.info("Certificates were fetched successfully")
+    except Exception:
+      raise Exception(f"Unable to retrieve certificate from {hostname}:{port}")
+
+    return f"SHA1 Fingerprint={fp}", readable_certificate
+
+  def write_certificate_to_file(self, fetched_certificate, cert_file_path, input_obj, force_overwrite_flag):
+    """
+    Method for writing certificate details to a file on the user's machine
+    """
+    if fetched_certificate is not None:
+      self._check_file_exists(cert_file_path, force_overwrite_flag)
+      with open(cert_file_path, 'w') as config_fp:
+          config_fp.write(fetched_certificate)
+
+      sys.stdout.write(f"Wrote certificate to {cert_file_path}\n")
+
+  def write_conjurrc(self, conjurrc_file_path, input_obj, force_overwrite_flag):
+    """
+    Method for writing the conjurrc configuration details needed to create a connection to Conjur
+    """
+    self._check_file_exists(conjurrc_file_path, force_overwrite_flag)
+
+    with open(conjurrc_file_path, 'w') as config_fp:
+      config_fp.write("---\n")
+      _pretty_print_object = {
+        'appliance_url': input_obj.appliance_url,
+        'account': input_obj.account,
+        'cert_file': input_obj.cert_file,
+        'plugins': input_obj.plugins
+      }
+
+      yaml.dump(_pretty_print_object, config_fp)
+
+    sys.stdout.write(f"Wrote configuration to {conjurrc_file_path}\n")
+
+  @staticmethod
+  def _check_file_exists(config_file, force_overwrite_flag):
```

Comment:
Rename func name it not checks if file exists.
Maybe name it to 
_overight_file_if_exists or something better you think of

Reply:
good call 👌